assets/src/components/detours/detourRouteSelectionPanel.tsx

```diff
@@ -63,27 +67,43 @@ export const DetourRouteSelectionPanel = ({
       <Panel.Body className="d-flex flex-column">
         <Panel.Body.ScrollArea className="d-flex flex-column">
           <section className="pb-3">
-            <h2 className="c-diversion-panel__h2" id={selectId}>
-              Choose route
-            </h2>
-            <FormSelect
-              aria-labelledby={selectId}
-              value={selectedRouteInfo.selectedRoute?.id}
-              onChange={(changeEvent) => {
-                onSelectRoute?.(
-                  allRoutes.find(
-                    (route) => route.id === changeEvent.target.value
-                  )
-                )
-              }}
+            <Form
+              // We're doing js validation, so disable HTML validation
+              noValidate={true}
             >
-              <option>Select a route</option>
-              {allRoutes.map((route) => (
-                <option key={route.id} value={route.id}>
-                  {route.name}
-                </option>
-              ))}
-            </FormSelect>
+              <Form.Group>
+                <h2 className="c-diversion-panel__h2" id={selectId}>
+                  Choose route
+                </h2>
+                <Form.Select
+                  // Never show the "valid" style
+                  isValid={false}
+                  // Show invalid when no route and after "submit"" is pressed
+                  isInvalid={isRouteInvalid}
+                  aria-invalid={isRouteInvalid}
+                  aria-errormessage={routeErrorId}
+                  aria-labelledby={selectId}
+                  value={selectedRouteInfo.selectedRoute?.id}
+                  onChange={(changeEvent) => {
+                    onSelectRoute?.(
+                      allRoutes.find(
+                        (route) => route.id === changeEvent.target.value
+                      )
+                    )
+                  }}
+                >
+                  <option value="">Select a route</option>
+                  {allRoutes.map((route) => (
+                    <option key={route.id} value={route.id}>
+                      {route.name}
+                    </option>
+                  ))}
+                </Form.Select>
```

Comment:
Hmmm.. a lot of plumbing was set up here that will get wiped by the combobox work, though I didn't try to set up this error state in the combobox yet. Not sure what to do with that sequencing yet

Reply:
After finishing review, I see that the plumbing wasn't high effort or anything, and all of it was a logical change given the combobox wasn't finished. I'll have to add more work to the combobox to ensure the same validation styling is applied